happi/audit.py

```diff
@@ -0,0 +1,137 @@
+"""
+This module contains functions and helpers for auditing the happi database.
+Checks are simple functions that take a happi.SearchResult a positional
+argument and returns None when successful.  When a check fails, it should throw
+an Exception with a helpful error message.  These exception messages will be
+caught and organized by the cli audit tool.
+"""
+import inspect
+from typing import Callable, List, Optional, Tuple
+
+from happi import SearchResult
+
+CLIENT_KEYS = ['_id', 'type', 'creation', 'last_edit']
+
+
+def check_instantiation(result: SearchResult) -> None:
+    """
+    Check if the device can be instantiated
+    Simply attempts to get the device from the happi.SearchResult
+
+    Subscription callback exceptions are silenced in ophyd, and will
+    not cause this check to fail.
+    """
+    _ = result.get()
```

Comment:
Nitpick, but why not just:
```suggestion
    result.get()
```

Reply:
That's a good question, seeing as I do that with the output of the check functions themselves...